app/src/main/java/org/wikipedia/page/PageFragment.java

```diff
@@ -995,9 +1080,27 @@ private void setupMessageHandlers() {
             }
         });
         bridge.addListener("back_link", (String messageType, JsonObject payload) -> {
+            int yOffset = DimenUtil.roundedPxToDp(webView.getScrollY());
+            int itemToSelect = 0;
+            if (tocHandler != null && tocHandler.getAdapter() != null) {
```

Comment:
There is a problem with finding the exact section, since the position of the section on the display is immaterial... @dbrant any suggestions?